src/PersistenceProviders/Proto.Persistence.Sqlite/SqliteProvider.cs

```diff
@@ -29,19 +48,24 @@ public SqliteProvider(string datasource = "states.db")
         {
             try
             {
-                using (var db = new SqlitePersistenceContext(_datasource))
+                using (var connection = new SqliteConnection(_connectionString))
                 {
-                    var items = db.Events
-                                    .Where(x => x.ActorName == actorName)
-                                    .Where(x => x.EventIndex <= inclusiveToIndex)
-                                    .ToList();
+                    connection.Open();
 
-                    db.RemoveRange(items);
-
-                    await db.SaveChangesAsync();
+                    using (var transaction = connection.BeginTransaction())
```

Comment:
Don't need a transaction over a single command.  Sqlite implicitly has a transaction on this.